src/tracks/labeltrack/ui/LabelGlyphHandle.cpp

```diff
@@ -310,6 +318,46 @@ bool LabelGlyphHandle::HandleGlyphDragRelease
          pTrack->SetLabel( hit.mMouseOverLabelRight, labelStruct );
       }
 
+      if (hit.mMouseOverLabel >= 0)
+      {
+          auto labelStruct = mLabels[hit.mMouseOverLabel];
+          if (!labelStruct.updated)
+          {
+              newSel = labelStruct.selectedRegion;
+
+              // IF the user clicked a label, THEN select all other tracks by Label
+              // do nothing if at least one other track is selected
+              auto& selectionState = SelectionState::Get(project);
+              auto& tracks = TrackList::Get(project);
+
+              bool done = tracks.Selected().any_of(
+                  [&](const Track* track) { return track != static_cast<Track*>(pTrack.get()); }
+              );
+
+              if (!done) {
+                  //otherwise, select all tracks
+                  for (auto t : tracks.Any())
+                      selectionState.SelectTrack(*t, true, true);
+              }
+
+              // Do this after, for its effect on TrackPanel's memory of last selected
+              // track (which affects shift-click actions)
+              selectionState.SelectTrack(*pTrack.get(), true, true);
+
+              // PRL: bug1659 -- make selection change undo correctly
+              updated = !ProjectAudioIO::Get(project).IsAudioActive();
+              
+              auto& view = LabelTrackView::Get(*pTrack);
+              view.SetNavigationIndex(hit.mMouseOverLabel);
+          }
+          else
+          {
+              labelStruct.updated = false;
+              pTrack->SetLabel(hit.mMouseOverLabel, labelStruct);
+              updated = true;
+          }
+      }
+
```

Comment:
This is a large block of moved code.  It used to happen in the click.  I'm not sure how I reach it in the debugger now.  Can the commit comment explain the purpose of this move?